test/Log/OpusLogTest.php

```diff
@@ -0,0 +1,211 @@
+        $instance->setAccessible(true);
+        $instance->setValue(null, null);
+        $instance->setAccessible(false);
+
+        $this->removeFolder($this->tempFolder);
+
+        parent::tearDown();
+    }
+
+    public function testSetPriority()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setPriority('debug');
+
+        $debugMessage = 'Debug level message from testSetPriority';
+        $opusLog->debug($debugMessage);
+        $content = $this->readLogFile('default.log');
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetPriorityWithNullPriority()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setPriority(null);
+
+        $debugMessage = 'Debug level message from testSetPriorityWithNullPriority';
+        $opusLog->debug($debugMessage);
+
+        $infoMessage = 'Info level message from testSetPriorityWithNullPriority';
+        $opusLog->info($infoMessage);
+
+        $content = $this->readLogFile('default.log');
+
+        $this->assertContains($infoMessage, $content);
+    }
+
+    public function testSetPriorityUnknownPriority()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->setExpectedException(\Exception::class, "No such priority found as TestPriority");
+
+        $opusLog->setPriority('TestPriority');
+    }
+
+    public function testGetPriority()
+    {
+        $opusLog = $this->getOpusLog();
+        $priority = $opusLog->getPriority();
+
+        $this->assertEquals('INFO', $priority);
+    }
+
+    public function testGetPriorityAfterSetPriority()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setPriority('debug');
+        $priority = $opusLog->getPriority();
+
+        $this->assertEquals('DEBUG', $priority);
+    }
+
+    public function testConvertPriorityToInt()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->assertEquals(\Zend_Log::DEBUG, $opusLog->convertPriorityToInt('debug'));
+        $this->assertEquals(\Zend_Log::CRIT, $opusLog->convertPriorityToInt('CRIT'));
+        $this->assertEquals(\Zend_Log::INFO, $opusLog->convertPriorityToInt('Info'));
+    }
+
+    public function testConvertPriorityToIntUnknownPriority()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->assertNull($opusLog->convertPriorityToInt('TestLevel'));
+    }
+
+    public function testConvertPriorityToString()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->assertEquals('INFO', $opusLog->convertPriorityToString(\Zend_Log::INFO));
+        $this->assertEquals('ERR', $opusLog->convertPriorityToString(\Zend_Log::ERR));
+        $this->assertEquals('EMERG', $opusLog->convertPriorityToString(\Zend_Log::EMERG));
+    }
+
+    public function testConvertPriorityToStringUnknownPriority()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->assertNull($opusLog->convertPriorityToString(10));
+    }
+
+    public function getOpusLog()
```

Comment:
Do we really need this function? What is the thought here? If we want it, for whatever reason, should it be public? Will somebody create an instance of `OpusLogTest` und call `getOpusLog`?

Reply:
A function for getting the test object can be useful, but do we really want to create that object in `setUp` in that case? Do we need the variable `$opusLog` in that case? There at least two options here, right now they are mixed and that is unnecessary.

Reply:
The idea is to get an object of `Log` class through `setUp()`. I will change `getOpusLog()` to `protected`.

Reply:
Why do we need to create the Log in setUp if we use this function?

Reply:
Creating and returning the log in `getOpusLog()` itself now.